lib/modules/ai_bot/commands/search_settings_command.rb

```diff
@@ -0,0 +1,85 @@
+#frozen_string_literal: true
+
+module DiscourseAi::AiBot::Commands
+  class SearchSettingsCommand < Command
+    class << self
+      def name
+        "search_settings"
+      end
+
+      def desc
+        "Will search through site settings and return top 20 results"
+      end
+
+      def parameters
+        [
+          Parameter.new(
+            name: "query",
+            description:
+              "comma delimited list of settings to search for (e.g. 'setting_1,setting_2')",
+            type: "string",
+            required: true,
+          ),
+        ]
+      end
+    end
+
+    def result_name
+      "results"
+    end
+
+    def description_args
+      { count: @last_num_results || 0, query: @last_query || "" }
+    end
+
+    INCLUDE_DESCRIPTIONS_MAX_LENGTH = 10
+    MAX_RESULTS = 200
+
+    def process(query:)
+      @last_query = query
+      @last_num_results = 0
+
+      terms = query.split(",").map(&:strip).map(&:downcase).reject(&:blank?)
```

Comment:
Maybe we can add  min length and early return? Pointless to allow people to use up CPU time searching for `a` :)

Reply:
I think a is OK, its not chewing up too much cpu... this is going to have to scan through everything no matter what you give it... no indexes.

Reply:
Yeah I just mean it's kind of pointless to even try with anything < 3 characters long...but if you think it's fine just leave it :+1: